src/stores/main/utils.ts

```diff
@@ -18,3 +19,250 @@ export function mapOAuth2UserToUser(user: IOAuth2User): ILoggedInUser {
     isAdmin,
   };
 }
+
+export function computePermissions(
+  ruleReviews: readonly [
+    namespace: string,
+    review: authorizationv1.ISelfSubjectRulesReview
+  ][]
+): IPermissionMap {
+  const permissions: IPermissionMap = {};
+
+  for (const [namespace, review] of ruleReviews) {
+    if (!review.status || review.status.incomplete) {
+      continue;
+    }
```

Comment:
I wonder if that isn't a race if that status is not created yet.

Reply:
Actually it seems that our type is incorrect on that one. The type in the API is not nullable. I'll update it.

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/apis/authorization/types.go#L166